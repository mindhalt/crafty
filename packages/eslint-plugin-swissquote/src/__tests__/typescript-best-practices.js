/* global describe, it, expect */

const { prepareCLIEngine, lint } = require("../../test_utils");
const engine = prepareCLIEngine(
  require("../best-practices"),
  require("../typescript-best-practices"),
  {
    parser: require.resolve("@typescript-eslint/parser"),
    env: { browser: true }
  }
);

it("Warns on console.log", () => {
  const result = lint(engine, 'console.log("Yeah");');

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toBe(1);
  expect(result.errorCount).toBe(0);
});

it("Doesn't warn on valid but badly formatted code", () => {
  const result = lint(
    engine,
    `

function test() { fetch("This is spartaaa"); }

test()

`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toBe(0, "no warnings");
  expect(result.errorCount).toBe(0, "no errors");
});

it("Uses sonar plugin", () => {
  const result = lint(
    engine,
    `
/* global openWindow, closeWindow, moveWindowToTheBackground */

function changeWindow(param: number) {
  if (param === 1) {
    openWindow();
  } else if (param === 2) {
    closeWindow();
  } else if (param === 1) { // Noncompliant
    moveWindowToTheBackground();
  }
}

`
  );

  expect(result.messages).toMatchSnapshot();
  expect(result.warningCount).toBe(0);
  expect(result.errorCount).toBe(2);
});
