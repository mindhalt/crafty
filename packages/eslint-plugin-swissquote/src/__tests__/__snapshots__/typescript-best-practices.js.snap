// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doesn't warn on valid but badly formatted code 1`] = `Array []`;

exports[`Uses sonar plugin 1`] = `
Array [
  Object {
    "column": 10,
    "endColumn": 22,
    "endLine": 3,
    "line": 3,
    "message": "'changeWindow' is defined but never used.",
    "nodeType": "Identifier",
    "ruleId": "@swissquote/swissquote/@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
  Object {
    "column": 14,
    "endColumn": 25,
    "endLine": 8,
    "line": 8,
    "message": "This branch duplicates the one on line 4",
    "nodeType": "BinaryExpression",
    "ruleId": "@swissquote/swissquote/sonarjs/no-identical-conditions",
    "severity": 2,
  },
]
`;

exports[`Warns on console.log 1`] = `
Array [
  Object {
    "column": 1,
    "endColumn": 12,
    "endLine": 1,
    "line": 1,
    "message": "Unexpected console statement.",
    "messageId": "unexpected",
    "nodeType": "MemberExpression",
    "ruleId": "no-console",
    "severity": 1,
  },
]
`;
